### PR TITLE
go: Require a TTL to be set for all outgoing calls.

### PR DESCRIPTION
**Pull request by prashantv**
r: @mmihic 
cc: @Raynos

### CODE REVIEW

**Comment by mmihic** [comment]
lgtm

File: golang/outbound.go [hunk 1]
```diff
@@ -46,9 +46,13 @@ func (c *Connection) beginCall(ctx context.Context, serviceName string, callOpti
 		return nil, err
 	}
 
-	deadline, _ := ctx.Deadline()
+	deadline, ok := ctx.Deadline()
```
Comment (Raynos) [inline golang/outbound.go]: consider defaulting a deadline ? In node we default you into 100ms if you do not set a timeout.

**Comment by Raynos** [comment]
lgtm.

**Comment by prashantv** [comment]
Keeping explicit as it's easy to decide on a timeout when creating a context.

**Pull request merged**